bundle/render/render_text_output.go

```diff
@@ -139,19 +158,74 @@ type RenderOptions struct {
 	RenderSummaryTable bool
 }
 
-// RenderTextOutput renders the diagnostics in a human-readable format.
-func RenderTextOutput(out io.Writer, b *bundle.Bundle, diags diag.Diagnostics, opts RenderOptions) error {
-	err := renderDiagnostics(out, b, diags)
+// RenderDiagnostics renders the diagnostics in a human-readable format.
+func RenderDiagnostics(out io.Writer, b *bundle.Bundle, diags diag.Diagnostics, opts RenderOptions) error {
+	err := renderDiagnosticsOnly(out, b, diags)
 	if err != nil {
 		return fmt.Errorf("failed to render diagnostics: %w", err)
 	}
 
 	if opts.RenderSummaryTable {
-		err = renderSummaryTemplate(out, b, diags)
-		if err != nil {
-			return fmt.Errorf("failed to render summary: %w", err)
+		if b != nil {
+			err = renderSummaryHeaderTemplate(out, b)
+			if err != nil {
+				return fmt.Errorf("failed to render summary: %w", err)
+			}
+			io.WriteString(out, "\n")
 		}
+		trailer := buildTrailer(diags)
+		io.WriteString(out, trailer)
 	}
 
 	return nil
 }
+
+func RenderSummary(ctx context.Context, out io.Writer, b *bundle.Bundle) error {
+	if err := renderSummaryHeaderTemplate(out, b); err != nil {
+		return err
+	}
+
+	var resourceGroups []ResourceGroup
+
+	for group, r := range b.Config.Resources.AllResources() {
+		resources := make([]ResourceInfo, 0, len(r))
+		for key, resource := range r {
+			resources = append(resources, ResourceInfo{
+				Key:  key,
+				Name: resource.GetName(),
+				URL:  resource.GetURL(),
+			})
+		}
+
+		if len(resources) > 0 {
+			capitalizedGroup := strings.ToUpper(group[:1]) + group[1:]
```

Comment:
This yields `Registered_models`.

You can add the capitalized plural to `config.ResourceDescription` and `config.SupportedResources`.

Reply:
That seems okay though? The design here seems to be to (mostly) follow YML conventions. But somehow add capitals.

Reply:
Either we do human-readable or the YAML keys, not a new hybrid (it doesn't surface anywhere else).

Reply:
Fixed in latest.